server/service/service_osquery.go

```diff
@@ -1121,6 +1121,8 @@ func (svc *Service) SubmitDistributedQueryResults(ctx context.Context, results f
 		if strings.HasPrefix(query, hostLabelQueryPrefix) {
```

Comment:
This seemed not quite right to me, and I did a little digging.

https://github.com/fleetdm/fleet/pull/420/files is where it was implemented. Based on my comment in https://github.com/fleetdm/fleet/pull/420#issuecomment-793421551, I think this was intended to make sure we load the host additional info when the _detail_ queries are updated, but it happened to have the side-effect of fixing the failing test.

I think the proper code here would be 

```suggestion
		if strings.HasPrefix(query, hostDetailQueryPrefix) {
```